### PR TITLE
Add ERC: NFTs Benefits Attachment via Metadata

### PR DESCRIPTION
**Pull request by asghaier76**
When opening a pull request to submit a new EIP, please use the suggested template: https://github.com/ethereum/EIPs/blob/master/eip-template.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing draft PRs.
 - The build passes.
 - Your GitHub username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

File: ERCS/erc-7950.md [hunk 1]
```diff
@@ -0,0 +1,400 @@
+---
+eip: ....
```
Comment (abcoathup) [inline ERCS/erc-7950.md:2]:
```suggestion
eip: 7554
```
Assigning next sequential EIP/ERC/RIP number.
Numbering changed to sequential from 7500 and is no longer the PR number.

Please also update the filename.

File: ERCS/erc-7950.md [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+title: NFT Benefits Attachment and Benefits Metadata Extension
```
Comment (abcoathup) [inline ERCS/erc-7950.md:3]:
```suggestion
title: Benefits attached to NFTs via metadata
```
Need to shorten the title.  Perhaps like the above suggestion.

File: ERCS/erc-7950.md (hunk 1 above)
Comment (asghaier76) [reply ERCS/erc-7950.md]: Updated to 7554 as well as filename

**Comment by asghaier76** [comment]
@abcoathup all comments have been addressed and ready for review

**Comment by abcoathup** [comment]
> @abcoathup all comments have been addressed and ready for review

Need to wait for an ERC editor to review.

File: ERCS/erc-7554.md [hunk 3]
```diff
@@ -0,0 +1,400 @@
...
+
+This standard aims to provide the capability to attach and manage benefits attached to NFTs using on-chain functionality and with a standardized benefits metadata that will make it easy for 3rd parties such as marketplaces, aggregators and data analytics actors to collect that information.
+ 
+The standard aims to specify an interface that will allow capturing the events triggered by attaching and disattaching benefits as well as the standard functions to allow querying all benefits attached to a single NFT or a collection.
+
+The use of NFT token-gating and benefits attachment has already been considered in many applications like the adoption of Shopify of the token-gating concept that can open the doors for many use cases including for example an online store offering a discount for customers who own a specific NFT; an airdrop to be limited to only owners of NFTs from a certain collection; a restaurant can offer certain coupons and discounts to some holders, and many others.
```
Comment (SamWilsn) [inline ERCS/erc-7554.md:28]: > [...] the adoption of Shopify of the token-gating concept that can open the doors for many use cases including for example an online store offering a discount for customers who own a specific NFT [...]

This bit is a bit hard to parse. I'd recommend rewording it.

File: ERCS/erc-7554.md [hunk 4]
```diff
@@ -0,0 +1,400 @@
...
+
+The contract owner MAY set the `attachBenefit` as payable to be able to collect fees in advance and on-chain from interested parties in attaching benefits to the collection tokens.
+
+The `BenefitAttached` event MUST be emitted when `attachBenefit` is called.
+
+The `BenefitUpdated` event MUST be emitted when `updateBenefit` is called.
+
+The `BenefitRemoved` event MUST be emitted when `removeBenefit` is called.
+
+The `supportsInterface` method MUST return `true` when called with `TBD`.
```
Comment (SamWilsn) [inline ERCS/erc-7554.md:170]:
```suggestion
The `supportsInterface` method MUST return `true` when called with `TBD`. <!-- TODO -->
```

If you use HTML-style comments, the linter will warn you if you attempt to go to review without removing them.

File: ERCS/erc-7554.md [hunk 5]
```diff
@@ -0,0 +1,400 @@
...
+}
+```
+
+Here is an example for a benefit attached to BAYC NFTs on token basis offered by AirOne Airlines to get one time access to their VIP Lounge in Dream City airport, that can be redeemed during the period Sep 20, 2023 to Dec 20, 2023 
+
+```json
+{
+    "title": "VIP Lounge Access",
+    "Description": "Provides one time access to AirOne Airlines VIP Lounge in Dubai airport",   
+    "imageURI": "https://airone.com/images/bayc-vip-lounge.png",
```
Comment (SamWilsn) [inline ERCS/erc-7554.md:288]: Please use designated example domains (eg. `example.com`) for your examples.

File: ERCS/erc-7554.md [hunk 6]
```diff
@@ -0,0 +1,400 @@
...
+  }
+}
+```
+
+Here is an example for a benefit attached to BAYC NFTs on token basis offered by AirOne Airlines to get one time access to their VIP Lounge in Dream City airport, that can be redeemed during the period Sep 20, 2023 to Dec 20, 2023 
+
+```json
+{
+    "title": "VIP Lounge Access",
+    "Description": "Provides one time access to AirOne Airlines VIP Lounge in Dubai airport",
```
Comment (SamWilsn) [inline ERCS/erc-7554.md:287]: Should this be `description` instead?

File: ERCS/erc-7554.md [hunk 7]
```diff
@@ -0,0 +1,400 @@
...
+
+The other important reason for defining this standard is to allow collection owners to have control on the possible value driven from attaching benefits to NFTs in their collections, this is for now is left in the hands of brands and other third party platforms for e-commerce, loyalty and token-gating to extract this value and tap into the commmunity that the collection owner has built. This approach can also stem more interest from collection owners to continue building more their communities with the possiblity of building business relations with brands and thrid party platforms.
+
+The collection owners in this case can be more incentivized to keep building a vibrant community with possible monetization scheme on these efforts, and that can also serve in increased valuation of the NFTs themselves.
```
Comment (SamWilsn) [inline ERCS/erc-7554.md:382]: This content all belongs in the motivation section, because it's justifying the proposal as a whole. The rationale section is meant for the reasoning behind choices made _within_ the proposal itself.

File: ERCS/erc-7554.md [hunk 8]
```diff
@@ -0,0 +1,400 @@
...
+
+## Backwards Compatibility
+
+This ERC can be fully compatible with [ERC-721](./eip-721.md) by adding the functions and events defined above which are not going to affect the original functionality of [ERC-721](./eip-721.md) compatible contracts.
+
+For already existing deployed [ERC-721](./eip-721.md) contracts, the collection owner can deploy a contract based on this standard that will provide the features mentioned in this standard, and to be able to define the original collection contract address this will require adding a function to query the contract address of the original collection.
+
+## Reference Implementation
+
+WIP
```
Comment (SamWilsn) [inline ERCS/erc-7554.md:392]:
```suggestion
<!--
## Reference Implementation

WIP
-->
```

This section is optional, so you can just comment it out.

File: ERCS/erc-7554.md [hunk 9]
```diff
@@ -0,0 +1,400 @@
...
+
+## Security Considerations
+
+WIP
```
Comment (SamWilsn) [inline ERCS/erc-7554.md:396]:
```suggestion
<!-- WIP -->
```

**Review by SamWilsn — COMMENTED** [review]
I'd recommend running your schema against the examples provided, to make sure they validate. I like https://www.jsonschemavalidator.net/
